src/Microsoft.IdentityModel.Tokens/Validation/Results/Details/ValidationError.cs

```diff
@@ -7,10 +7,15 @@
 
 namespace Microsoft.IdentityModel.Tokens
 {
+    internal interface ISecurityTokenException
+    {
+        void SetValidationError(ValidationError validationError);
+    }
+
     /// <summary>
     /// Contains information so that Exceptions can be logged or thrown written as required.
     /// </summary>
-    internal class ValidationError
+    public class ValidationError
```

Comment:
Do we want to move this to public at this point?

Reply:
We will need to make this public for the time when we make the new validation path public.
Right now, in order to implement the interface I needed to make the type public, but I kept the constructors and methods internal to prevent it from being used at all.
With the current changes the type "exists" but is not usable outside of the library.